nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-kubernetes-bundle/nifi-framework-kubernetes-state-provider/src/main/java/org/apache/nifi/kubernetes/state/provider/KubernetesConfigMapStateProvider.java

```diff
@@ -99,6 +118,12 @@ public void initialize(final StateProviderInitializationContext context) {
         this.logger = context.getLogger();
         this.kubernetesClient = getKubernetesClient();
         this.namespace = new ServiceAccountNamespaceProvider().getNamespace();
+
+        String configMapNamePrefix = context.getProperty(CONFIG_MAP_NAME_PREFIX).isSet() ? context.getProperty(CONFIG_MAP_NAME_PREFIX).getValue() : null;
+        CONFIG_MAP_NAME_FORMAT = configMapNamePrefix != null
+            ? String.format("%s-%s-%%s", configMapNamePrefix, CONFIG_MAP_CORE_NAME) : String.format("%s-%%s", CONFIG_MAP_CORE_NAME);
+        CONFIG_MAP_NAME_PATTERN = Pattern.compile(configMapNamePrefix != null
+            ? String.format("^%s-%s-(.+)$", configMapNamePrefix, CONFIG_MAP_CORE_NAME) : String.format("^%s-(.+)$", CONFIG_MAP_CORE_NAME));
```

Comment:
There is still a bit of repetition here that would be helpful to streamline, given the importance of proper formatting. The ternary expression is also difficult to read. I recommend the `PATTERN_FORMAT` as a static value with `^%snifi-component-(.+)$` and then supplying either an empty string, or the configured prefix to build the final Pattern. Declaring the intermediate values could also make this more readable.

Reply:
I made the change. I let you confirm that it was what you wanted.

Reply:
Thanks for making the adjustments @juldrixx, the latest version looks much cleaner!